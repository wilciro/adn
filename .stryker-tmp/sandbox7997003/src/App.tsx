// @ts-nocheck
// React
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import React, { FC } from 'react';
import { BrowserRouter } from 'react-router-dom'; // Context

import LayoutHeader from 'components/Layout/Header';
import LayoutFooter from 'components/Layout/Footer';
import LayoutBody from 'components/Layout/Body';
import CustomRoutes from 'components/CustomRoutes';
import ErrorBoundary from 'config/errors/GlobalErrorBoundary';
import { SessionProvider } from 'context/SessionContext';
import { MantineProvider } from '@mantine/core';

const App: FC = () => {
  if (stryMutAct_9fa48("0")) {
    {}
  } else {
    stryCov_9fa48("0");
    return <MantineProvider theme={stryMutAct_9fa48("1") ? {} : (stryCov_9fa48("1"), {
      colorScheme: stryMutAct_9fa48("2") ? "" : (stryCov_9fa48("2"), 'light'),
      colors: stryMutAct_9fa48("3") ? {} : (stryCov_9fa48("3"), {
        brand: stryMutAct_9fa48("4") ? [] : (stryCov_9fa48("4"), [stryMutAct_9fa48("5") ? "" : (stryCov_9fa48("5"), '#10d35f'), stryMutAct_9fa48("6") ? "" : (stryCov_9fa48("6"), '#10d35f'), stryMutAct_9fa48("7") ? "" : (stryCov_9fa48("7"), '#10d35f'), stryMutAct_9fa48("8") ? "" : (stryCov_9fa48("8"), '#10d35f'), stryMutAct_9fa48("9") ? "" : (stryCov_9fa48("9"), '#10d35f'), stryMutAct_9fa48("10") ? "" : (stryCov_9fa48("10"), '#10d35f'), stryMutAct_9fa48("11") ? "" : (stryCov_9fa48("11"), '#10d35f'), stryMutAct_9fa48("12") ? "" : (stryCov_9fa48("12"), '#10d35f'), stryMutAct_9fa48("13") ? "" : (stryCov_9fa48("13"), '#10d35f'), stryMutAct_9fa48("14") ? "" : (stryCov_9fa48("14"), '#10d35f')])
      }),
      primaryColor: stryMutAct_9fa48("15") ? "" : (stryCov_9fa48("15"), 'brand')
    })} withGlobalStyles withNormalizeCSS>
      <BrowserRouter>
        <ErrorBoundary>
          <SessionProvider>
            <LayoutHeader />
            <LayoutBody>
              <CustomRoutes />
            </LayoutBody>
            <LayoutFooter />
          </SessionProvider>
        </ErrorBoundary>
      </BrowserRouter>
    </MantineProvider>;
  }
};

export default App;