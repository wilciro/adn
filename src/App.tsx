// React
import React, { FC } from 'react';
import { BrowserRouter } from 'react-router-dom';

// Context
import LayoutHeader from 'components/Layout/Header';
import LayoutFooter from 'components/Layout/Footer';
import LayoutBody from 'components/Layout/Body';
import CustomRoutes from 'components/CustomRoutes';
import ErrorBoundary from 'config/errors/GlobalErrorBoundary';
import { SessionProvider } from 'context/SessionContext';
import { MantineProvider } from '@mantine/core';

const App: FC = () => {
  return (
    <MantineProvider
      theme={{
        colorScheme: 'light',
        colors: {
          brand: [
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
            '#10d35f',
          ],
        },
        primaryColor: 'brand',
      }}
      withGlobalStyles
      withNormalizeCSS
    >
      <BrowserRouter>
        <ErrorBoundary>
          <SessionProvider>
            <LayoutHeader />
            <LayoutBody>
              <CustomRoutes />
            </LayoutBody>
            <LayoutFooter />
          </SessionProvider>
        </ErrorBoundary>
      </BrowserRouter>
    </MantineProvider>
  );
};

export default App;
