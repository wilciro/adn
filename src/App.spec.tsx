import React from 'react';
import {
  act,
  fireEvent,
  render,
  RenderOptions,
  screen,
  waitFor,
} from '@testing-library/react';
import App from './App';

describe('App tests', () => {
  test('should match snapshot App', async () => {
    let container: RenderOptions<
      typeof import('@testing-library/dom/types/queries'),
      HTMLElement,
      HTMLElement
    >;
    await act(async () => {
      render(<App />, container);
    });

    expect(screen.getAllByTestId('header-landing')).toHaveLength(1);
  });
  test('should go to login', async () => {
    let container: RenderOptions<
      typeof import('@testing-library/dom/types/queries'),
      HTMLElement,
      HTMLElement
    >;
    await act(async () => {
      render(<App />, container);
    });

    fireEvent.click(screen.getByTestId('header-login-link'));
    await waitFor(() => {
      expect(screen.getAllByTestId('title-login')).toHaveLength(1);
    });

    fireEvent.click(screen.getByTestId('link-to-about'));
    await waitFor(() => {
      expect(screen.getAllByTestId('about-header')).toHaveLength(1);
    });

    fireEvent.click(screen.getByTestId('go-landing-btn'));
    await waitFor(() => {
      expect(screen.getAllByTestId('header-landing')).toHaveLength(1);
    });
  });
});
