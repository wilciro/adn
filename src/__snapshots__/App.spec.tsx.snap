// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match snapshot App 1`] = `
<div>
  <header
    class="sc-bczRLJ eJfard"
  >
    <div
      class="sc-gsnTZi hpkuKe"
    >
      <button
        class="mantine-ActionIcon-hover mantine-ActionIcon-root mantine-1vd7fg7"
        data-testid="btn-nav-menu"
        title="Menú"
        type="button"
      >
        <svg
          class="icon icon-tabler icon-tabler-menu-2"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="6"
            y2="6"
          />
          <line
            x1="4"
            x2="20"
            y1="12"
            y2="12"
          />
          <line
            x1="4"
            x2="20"
            y1="18"
            y2="18"
          />
        </svg>
      </button>
      <img
        alt="Logo"
        src="/assets/192.png"
      />
    </div>
    <div
      class="sc-gsnTZi hpkuKe"
    >
      <a
        href="/login"
        id="header-login-link"
      >
        <button
          class="mantine-Button-filled mantine-Button-root mantine-14wtse4"
          type="button"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Ingresar
            </span>
          </div>
        </button>
      </a>
    </div>
  </header>
  <nav
    class="sc-dkzDqf JmMFC"
    data-testid="nav-menu"
  >
    <button
      class="sc-eCYdqJ kgffIQ"
    >
      X
    </button>
    <ul
      class="sc-hKMtZM jUMTUe"
    >
      <li>
        <a
          href="/"
        >
          Inicio
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          Acerca de la app
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="sc-gKXOVf leHwAx"
  >
    <div
      class="mantine-yv0or0"
    >
      <section
        class="sc-iBkjds fXsXJJ"
      >
        <h2>
          Total: 
          $ 0,00
        </h2>
      </section>
      <h2>
        Solicitud de cita veterinaria
      </h2>
      <p>
        Diligencia todos los campos para asignar la cita
      </p>
      <div
        class="mantine-1ne3752"
      />
      <form>
        <div>
          <div
            class="mantine-TextInput-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-TextInput-label"
              for="mantine-ayvqfhb5o"
              id="mantine-ayvqfhb5o-label"
            >
              Nombre del dueño
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-162fluf"
                data-testid="ownerName"
                id="mantine-ayvqfhb5o"
                placeholder="Ingrese el nombre del dueño"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-TextInput-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-TextInput-label"
              for="mantine-nxwavvcds"
              id="mantine-nxwavvcds-label"
            >
              Documento del dueño
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-162fluf"
                data-testid="ownerDocument"
                id="mantine-nxwavvcds"
                placeholder="Ingrese el documento del dueño"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-TextInput-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-TextInput-label"
              for="mantine-9vnj71ljf"
              id="mantine-9vnj71ljf-label"
            >
              Nombre de la mascota
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-162fluf"
                data-testid="petName"
                id="mantine-9vnj71ljf"
                placeholder="Ingrese el nombre de la mascota"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-NumberInput-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-NumberInput-label"
              for="mantine-bbfvbccic"
              id="mantine-bbfvbccic-label"
            >
              Edad de la mascota
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-NumberInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-NumberInput-wrapper mantine-12sbrde"
            >
              <input
                aria-invalid="false"
                class="mantine-NumberInput-defaultVariant mantine-NumberInput-input mantine-4zbs7r"
                data-testid="petAge"
                id="mantine-bbfvbccic"
                inputmode="numeric"
                placeholder="Ingrese la edad de la mascota"
                required=""
                step="1"
                type="text"
                value="0"
              />
              <div
                class="mantine-1cwybh2 mantine-NumberInput-rightSection"
              >
                <div
                  class="mantine-p0axji mantine-NumberInput-rightSection"
                >
                  <button
                    aria-hidden="true"
                    class="mantine-NumberInput-control mantine-NumberInput-controlUp mantine-jbog3w"
                    tabindex="-1"
                    type="button"
                  />
                  <button
                    aria-hidden="true"
                    class="mantine-NumberInput-control mantine-NumberInput-controlDown mantine-xb3ria"
                    tabindex="-1"
                    type="button"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-Select-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-Select-label"
              for="mantine-bfpqvdj91"
              id="mantine-bfpqvdj91-label"
            >
              Tipo de mascota
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-Select-required"
              >
                 *
              </span>
            </label>
            <div
              aria-controls="mantine-bfpqvdj91"
              aria-expanded="false"
              aria-haspopup="listbox"
              role="combobox"
              tabindex="-1"
            >
              <input
                type="hidden"
                value=""
              />
              <div
                class="mantine-Select-wrapper mantine-12sbrde"
              >
                <input
                  aria-autocomplete="list"
                  aria-invalid="false"
                  autocomplete="nope"
                  class="mantine-Select-defaultVariant mantine-Select-input mantine-1sxescn"
                  data-mantine-stop-propagation="false"
                  data-testid="petType"
                  id="mantine-bfpqvdj91"
                  placeholder="Seleccione el tipo de mascota"
                  readonly=""
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="mantine-uquezq mantine-Select-rightSection"
                >
                  <svg
                    data-chevron="true"
                    fill="none"
                    height="18"
                    style="color: rgb(134, 142, 150);"
                    viewBox="0 0 15 15"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-DatePicker-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-DatePicker-label"
              for="mantine-lyo0fkl7y"
              id="mantine-lyo0fkl7y-label"
            >
              Fecha de la cita
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-DatePicker-required"
              >
                 *
              </span>
            </label>
            <div>
              <div
                class="mantine-2uofp0 mantine-DatePicker-wrapper"
              >
                <div
                  class="mantine-DatePicker-wrapper mantine-12sbrde"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="mantine-DatePicker-defaultVariant mantine-DatePicker-input mantine-DatePicker-input mantine-uqznwk"
                    data-testid="date"
                    id="mantine-lyo0fkl7y"
                    name="date"
                    placeholder="Seleccione la fecha de la cita"
                    readonly=""
                    required=""
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="mantine-TimeInput-root mantine-1m3pqry"
          >
            <label
              class="mantine-1w07r5r mantine-TimeInput-label"
              for="mantine-6iklhm5fh"
              id="mantine-6iklhm5fh-label"
            >
              Hora de la cita
              <span
                aria-hidden="true"
                class="mantine-1m203yh mantine-TimeInput-required"
              >
                 *
              </span>
            </label>
            <div
              class="mantine-TimeInput-wrapper mantine-12sbrde"
            >
              <div
                aria-invalid="false"
                class="mantine-TimeInput-defaultVariant mantine-TimeInput-input mantine-162fluf"
                data-testid="time"
                placeholder="Seleccione la hora de la cita"
                required=""
              >
                <div
                  class="mantine-x78oh3 mantine-TimeInput-controls"
                >
                  <input
                    class="mantine-TimeInput-timeInput mantine-ix2kpu"
                    id="mantine-6iklhm5fh"
                    inputmode="numeric"
                    placeholder="--"
                    type="text"
                    value="01"
                  />
                  <div
                    class="mantine-Text-root mantine-1hwmn4h"
                    style="line-height: 1;"
                  >
                    :
                  </div>
                  <input
                    class="mantine-TimeInput-timeInput mantine-ix2kpu"
                    inputmode="numeric"
                    placeholder="--"
                    type="text"
                    value="31"
                  />
                  <input
                    class="mantine-kolqo2"
                    placeholder="am"
                    type="text"
                    value="pm"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        class="mantine-1ne3752"
      />
      <button
        class="mantine-Button-filled mantine-Button-root mantine-14wtse4"
        data-testid="register-btn"
        type="button"
      >
        <div
          class="mantine-3xbgk5 mantine-Button-inner"
        >
          <span
            class="mantine-qo1k2 mantine-Button-label"
          >
            Registrar
          </span>
        </div>
      </button>
    </div>
  </div>
  <header
    class="sc-jSMfEi dFwEwe"
  >
    Ceiba software 2022
  </header>
</div>
`;
